contracts/V1/StateChannelDiamondProxy/StateSnapshotFacet.sol

```diff
@@ -0,0 +1,217 @@
+        }
+
+        //Already on latest fork, progress state within it
+        if (!hasLatestDisputeProof && hasMilestoneProof) {
+            _handleForkProgress(
+                channelId,
+                updateStateSnapshotStruct.stateSnapshot,
+                updateStateSnapshotStruct.exitChannelBlocks,
+                updateStateSnapshotStruct.milestoneProofs.value
+            );
+            _updateStateSnapshot(channelId, newStateSnapshot);
+        }
+
+        //Teleport to latest fork and then progress state within it
+        if (hasLatestDisputeProof && hasMilestoneProof) {
+            _handleForkTeleport(
+                channelId,
+                updateStateSnapshotStruct,
+                newStateSnapshot
+            );
+            _handleForkProgress(
+                channelId,
+                updateStateSnapshotStruct.stateSnapshot,
+                updateStateSnapshotStruct.exitChannelBlocks,
+                updateStateSnapshotStruct.milestoneProofs.value
+            );
+            _updateStateSnapshot(channelId, newStateSnapshot);
+        }
+    }
+
+    function _handleForkTeleport(
+        bytes32 channelId,
+        UpdateStateSnapshotStruct memory updateStateSnapshotStruct,
+        StateSnapshot memory newStateSnapshot
+    ) internal {
+        // TODO: Implement teleportation to latest fork using dispute proof
+        // This should verify the dispute proof and update the state snapshot to the new fork
+    }
+
+    function _handleForkProgress(
+        bytes32 channelId,
+        StateSnapshot memory newStateSnapshot,
+        ExitChannelBlock[] memory exitChannelBlocks,
+        ForkMilestoneProof[] memory milestoneProofs
+    ) internal {
+        StateSnapshot memory currentStateSnapshot = stateSnapshots[channelId];
+
+        // Only process if both snapshots are for the latest fork
+        require(
+            newStateSnapshot.forkCnt == getDisputeLength(channelId) &&
+                stateSnapshots[channelId].forkCnt ==
+                getDisputeLength(channelId),
+            "StateSnapshot is not valid"
+        );
+
+        // for each exit channel block, there must be a milestone proof
+        require(
+            milestoneProofs.length == exitChannelBlocks.length,
+            "Milestone proofs length must match exit channel blocks length"
+        );
+
+        require(
+            exitChannelBlocks[0].previousBlockHash ==
+                currentStateSnapshot.latestExitChannelBlockHash,
+            "First Exit block not linked to previous block"
+        );
+
+        require(
+            newStateSnapshot.latestExitChannelBlockHash ==
+                keccak256(
+                    abi.encode(exitChannelBlocks[exitChannelBlocks.length - 1])
+                ),
+            "Last Exit block not linked to new state snapshot"
+        );
+
+        // Get current participants from state snapshot
+        address[] memory currentParticipants = getSnapshotParticipants(
```

Comment:
Re-use the function linked above - this is duplicating the logic - participants shouldn't be used here at all, not even when processing exits - even the exits are not interpreted here